mesonbuild/modules/gnome.py

```diff
@@ -218,7 +263,77 @@ def compile_resources(self, state, args, kwargs):
             h_kwargs['install_dir'] = kwargs.get('install_dir',
                                                  state.environment.coredata.get_builtin_option('includedir'))
         target_h = GResourceHeaderTarget(args[0] + '_h', state.subdir, state.subproject, h_kwargs)
-        rv = [target_c, target_h]
+
+        if gresource_ld_binary:
```

Comment:
Please turn this into its own function. The functionality is fairly isolated and this function is getting way too long already.